pkg/collector/corechecks/system/disk/diskv2/disk_windows.go

```diff
@@ -0,0 +1,135 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build windows
+
+package diskv2
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	win "golang.org/x/sys/windows"
+	"regexp"
+	"strings"
+	"unsafe"
+)
+
+func compileRegExp(expr string) (*regexp.Regexp, error) {
+	iExpr := fmt.Sprintf("(?i)%s", expr)
+	return regexp.Compile(iExpr)
+}
+
+func (c *Check) fetchAllDeviceLabelsFromLsblk() error {
+	return nil
+}
+
+func (c *Check) fetchAllDeviceLabelsFromBlkidCache() error {
+	return nil
+}
+
+func (c *Check) fetchAllDeviceLabelsFromBlkid() error {
+	return nil
+}
+
+var mpr = win.NewLazySystemDLL("mpr.dll")
+var procWNetAddConnection2W = mpr.NewProc("WNetAddConnection2W")
+
+type netResource struct {
+	Scope       uint32
+	Type        uint32
+	DisplayType uint32
+	Usage       uint32
+	localName   *uint16
+	remoteName  *uint16
+	comment     *uint16
+	provider    *uint16
+}
+
+// RemotePath constructs the remote path based on the mount type.
+// It converts the Type to uppercase to ensure case-insensitive evaluation.
+func (m mount) RemotePath() (string, string) {
```

Comment:
Nitpick comment: The `mount` struct in the Linux version does not have the `RemotePath` function, but in the windows it does. Would it be better to create a private function inside `disk_windows.go` for extracting the remote instead of having two different structs base on the OS?